# **Municipios Región Caribe**

## **Librerías y modulos necesarios**

In [4]:
import warnings
import pandas as pd
from unidecode import unidecode
warnings.filterwarnings("ignore")
from funciones.rename_nivel import *

In [5]:
column_types = {"Código Departamento": "object", "Código Municipio": "object", "Código Centro Poblado": "object", "Nombre Departamento": "object", "Nombre Municipio": "object",
    "Nombre Centro Poblado	": "object", "Tipo": "object"}
url = "https://raw.githubusercontent.com/sePerezAlbor/Data/refs/heads/main/Divipola.csv"
data_divipola = pd.read_csv(url, delimiter = ",", na_values = [" "], dtype = column_types)

In [6]:
data_divipola["Nombre Departamento"] = data_divipola["Nombre Departamento"].apply(lambda x: unidecode(x))
data_divipola["Nombre Municipio"] = data_divipola["Nombre Municipio"].apply(lambda x: unidecode(x))
data_divipola = data_divipola[data_divipola["Nombre Departamento"].isin(regiones['CARIBE'])]

In [7]:
data_municipios = data_divipola[['Código Municipio', 'Nombre Municipio']]
data_municipios.drop_duplicates(subset=[ 'Código Municipio', 'Nombre Municipio'], inplace=True)
data_municipios.rename(columns={'Código Municipio': 'COD_MUNI', 'Nombre Municipio': 'MUNICIPIO'}, inplace=True)
data_municipios['COD_MUNI'] = data_municipios['COD_MUNI'].astype(str).str.zfill(5)

In [8]:
data_municipios.loc[len(data_municipios)] = ['88001', 'SAN ANDRÉS']
data_municipios.loc[len(data_municipios)] = ['88564', 'PROVIDENCIA']

In [9]:
data_municipios

COD_MUNI         MUNICIPIO
836     08001      BARRANQUILLA
839     08078           BARANOA
843     08137  CAMPO DE LA CRUZ
845     08141        CANDELARIA
848     08296            GALAPA
...       ...               ...
7629    70771             SUCRE
7680    70820  SANTIAGO DE TOLU
7686    70823        TOLU VIEJO
195     88001        SAN ANDRÉS
196     88564       PROVIDENCIA

[197 rows x 2 columns]

In [10]:
data_municipios.to_csv('data_processed/data_municipios_name.csv', index=False)

En total hay 197 municipios en la región Caribe